design/dedicated-osd-pod.md

```diff
@@ -23,22 +23,34 @@ A more comprehensive discussions can be found at [this issue](https://github.com
 
 ## Proposal   
 
-We propose the following OSD startup change.
+We propose the following change to address the limitations.
+
+### Terms
+
+- OSD Discovery. A process that is given a device or directory path filter and use the filter to discover devices on the host. The process updates the OSD orchestration Configmap with the device paths and mark the orchestration status as "DeviceDiscovered". The process is created by Rook Operator and runs in a Kubernetes deployment on the storage node. This process is storage backend agnostic. Currently `rook osd` command implements this functionality.
+
+- OSD Provisioner. A process that is given device or directory paths upon start and make backend specific storage types. For now, the process prepare Ceph OSDs. Currently `rook osd` command implements this functionality.
```

Comment:
- "For now": Is there something else it would prepare in the future? I'm not sure there is.
- Let's state that it's a K8s batch job that is run once and exits